### ISSUE TITLE
How to insert a text at the beginning of a new docx page. 

### ISSUE THREAD

**Issue by prathyush-python** [python-openxml/python-docx#1245]
This cannot be a header or a footer. It's a different text for each page and has to be added at the beginning of the page. Essentially how to identify the beginning of a new page when you only have one section. There are multiple pages but only one section.

**Comment by scanny** [comment]
Sounds like you have two challenges:

1. how to identify page-break locations
2. how to insert a paragraph at an arbitrary location in the document

For detecting page-break locations you will need to look for `w:lastRenderedPageBreak` elements and hope that the docx client (e.g. Word, LibreOffice, GoogleDocs) wrote those for you. Word does, others you'll have to see.

The second challenge is handled by this, but only when on a paragraph boundary: 
https://python-docx.readthedocs.io/en/latest/api/text.html#docx.text.paragraph.Paragraph.insert_paragraph_before

One of the challenges is that pages do not in general break on paragraph boundaries. If the paragraph is 10 lines long and the page only has 5 lines of space left, half of it appears on the current page and half on the other. So you'll need to decide what to do in that case. `w:lastRenderedPageBreak` elements appear within a run, with text before and after in the general case.

You can detect paragraphs containing rendered page-breaks like this:
```python
paras_with_rpbs = [
    Paragraph(p, document)
    for p in document.element.xpath("./w:body/w:p[w:r/w:lastRenderedPageBreak]")
]

# -- verify those are the paragraphs you expect --
for para in paras_with_rpbs:
    print(f"'{para.text}'\n")
```

The challenge then is how you want to split those paragraphs, if that's what you decide to do. You're probably going to need to understand XPath for this job. Or I suppose you could step through the paragraph child elements.

**Comment by prathyush-python** [comment]
"I have a 4-page document and I use Microsoft Word. On the last page, there is a table in the middle of the page. Additionally, on this same page, there is text at the beginning (not in the header). I am using the python-docx library to insert data into the table and add rows as needed. Occasionally, when there is too much data, the table spills over onto page numbers 5, 6, 7, and so on. In such cases, I would like to ensure that the text at the beginning of page 4 is replicated at the beginning of pages 5, 6, 7, and so forth." That's why I am trying to figure out how to identify a new page when looping through paragraphs so I can may be use insert_before_paragraph to do it. Is there a way to do this ?

**Comment by scanny** [comment]
Ah, okay, so you'll need to split the table then, not the paragraph, since that's what's spanning pages.

A possibly better option is this feature of Word which can repeat part of the table on each page-break: https://support.microsoft.com/en-au/office/repeat-table-header-on-subsequent-pages-2ff677e0-3150-464a-a283-fa52794b4b41

First step would be to figure out if you can do that by hand using Word directly as an example, and if that suits, then figure out how to do it with the code.

**Comment by prathyush-python** [comment]
![image](https://github.com/python-openxml/python-docx/assets/45636342/c18e2479-a6fd-4b36-af7b-bb730ea19206)


It's not just the table; there's also some minor content located beneath the table. Sometimes, when the table becomes long enough, the remaining content spills onto the next page. To sign and add a name on the last page, I have no way of identifying the title of the document, which is represented by the text 'ABCD-SDAS...' at the top of the first page (Assuming I print all these to manual papers.). And Ideally things should be done via the code way so that I do not have to reopen the word and fill out stuff.

**Comment by scanny** [comment]
Yeah, you're going to need to make separate tables. You can't put paragraphs in the middle of a table.
